bcgov/PIMS
# PIMS-1854 Snowplow

## 🎯 Summary

  
[PIMS-1854](https://citz-imb.atlassian.net/browse/PIMS-1854?atlOrigin=eyJpIjoiNGZmMGMxZWI4ZWVjNDZiM2JkN2EyM2E5ZmFjNWNjMWYiLCJwIjoiaiJ9)


## Changes
- Adds Snowplow analytics code
- Currently tracking:
  - page views
  - map searches
  - table searches
  - errors
  - page activity

There's a few schema changes I need to ask GDX for:
Map Search
```ts
interface data {
  pid: number,
  pin: number,
  address: string,
  property_name: string,
  agencies: string[],
  administrative_areas: string[],
  regional_districts: string[],
  classifications: string[],
  property_types: string[],
  project_statuses: string[],
  polygon: string,
},
```

Table Search:
```ts
interface data {
    name: string,
    columnSortValue: string,
    columnSortName: string,
    columnFilterName: string,
    columnFilterValue: string,
    columnFilterMode: string,
    keywordFilter: string,
    selectFilter: string,
    pageSize: number,
  },
```

~Waiting for a response from GDX so they can share our custom schemas. Also would like access to the dashboard so we can see how they were presenting the previously collected data. Can make decisions on what data to send after that.~

~Trying to use the npm packages instead of the JS version in old PIMS. Both have examples on the documentation site.
https://docs.snowplow.io/docs/collecting-data/collecting-from-own-applications/javascript-trackers/web-tracker/tracker-setup/~

Went back to the JS version from old PIMS because of CORS issues.

## Testing
- Have the extension Snowplow Inspector installed in your browser.
- View it in the dev tools
- Navigate through PIMS, triggering the snowplow events to see them in the inspector.


## 🔰 Checklist


> - I have performed a self-review of my code.
> - I have commented my code, particularly in hard-to-understand areas.
> - I have made corresponding changes to the documentation where required.
> - I have tested my changes to the best of my ability.
> - My changes generate no new warnings.
